### PR TITLE
Added listtransactions RPC

### PR DESCRIPTION
**Pull request by DennisAMenace**
Task notes:

A little different from listtransactions (0.20.0 RPC)

- The result does not have all of the data that the (0.20.0 RPC) provides, but I think we can keep adding to this later in more PR's as needed.

### CODE REVIEW

**Comment by dangershony** [comment]
> label (AKA account in previous versions of BTC Core) was depreciated in 0.14 so I added this parameter for backwards compatibility, but it does nothing.

What backwards compatibility needs that and cant we just fix it there?
I am afriad we will carry this tech debt for a long tie if we put it back

**Comment by DennisAMenace** [comment]
> > label (AKA account in previous versions of BTC Core) was depreciated in 0.14 so I added this parameter for backwards compatibility, but it does nothing.
> 
> What backwards compatibility needs that and cant we just fix it there?
> I am afriad we will carry this tech debt for a long tie if we put it back

No problem, Done. I added this in to allow the selection of the account in the last commit.

File: src/External/NBitcoin/ChainIndexer.cs [hunk 1]
```diff
@@ -179,7 +179,7 @@ public virtual IEnumerable<ChainedHeader> EnumerateAfter(ChainedHeader block)
         }
 
         /// <summary>
-        /// TODO: Make this internal when the component moves to Stratis.Bitcoin
+        /// TODO: Make this internal when the component moves to Blockcore
```
Comment (dangershony) [inline src/External/NBitcoin/ChainIndexer.cs:182]: ah brilliant job

File: src/Features/Blockcore.Features.Notifications/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+﻿using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Blockcore.Features.Notifications.Tests")]
```
Comment (dangershony) [inline src/Features/Blockcore.Features.Notifications/AssemblyInfo.cs:3]: check with @sondreb he solved this on the property files I think

File: src/Tests/Blockcore.IntegrationTests/GlobalSuppressions.cs [hunk 3]
```diff
@@ -4,14 +4,14 @@
...
-[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "<Pending>", Scope = "member", Target = "~M:Stratis.Bitcoin.IntegrationTests.Miners.ProofOfStakeMintCoinsSpecification.pos_node_wallet_has_earned_coins_through_staking")]
-[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "<Pending>", Scope = "member", Target = "~M:Stratis.Bitcoin.IntegrationTests.Miners.ProofOfStakeMintCoinsSpecification.pow_wallet_broadcasts_tx_of_million_coins_and_pos_wallet_receives")]
-[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "<Pending>", Scope = "member", Target = "~M:Stratis.Bitcoin.IntegrationTests.SmartContracts.SmartContractMinerTests.SmartContract_ReferencingInputInSameBlock~System.Threading.Tasks.Task")]
\ No newline at end of file
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "<Pending>", Scope = "member", Target = "~M:Blockcore.IntegrationTests.Miners.ProofOfStakeMintCoinsSpecification.a_proof_of_stake_node_with_wallet")]
```
Comment (dangershony) [inline src/Tests/Blockcore.IntegrationTests/GlobalSuppressions.cs:7]: ah good good I cant believe I missed that

File: src/Features/Blockcore.Features.Notifications/AssemblyInfo.cs (hunk 2 above)
Comment (DennisAMenace) [reply src/Features/Blockcore.Features.Notifications/AssemblyInfo.cs]: Great find, I confirmed with Sondre and resolved this

**Comment by dangershony** [comment]
@DennisAMenace please just fix the conflict and we push this

**Pull request merged**